Formula/shairport-sync.rb

```diff
@@ -24,34 +25,41 @@ class ShairportSync < Formula
   depends_on "autoconf" => :build
   depends_on "automake" => :build
   depends_on "pkg-config" => :build
-  depends_on "libao"
   depends_on "libconfig"
   depends_on "libdaemon"
   depends_on "libsoxr"
   depends_on "openssl@1.1"
   depends_on "popt"
-  depends_on "pulseaudio"
+
+  on_macos do
+    depends_on "libao"
+  end
+
+  on_linux do
+    depends_on "alsa-lib"
+    depends_on "avahi"
+    depends_on "pulseaudio"
+  end
```

Comment:
On macOS, `libao` is lower quality option that was probably being used. Shouldn't be needed on Linux as it has better options.

On Linux:
- `alsa-lib` is recommendation - https://github.com/mikebrady/shairport-sync#more-about-shairport-sync
   > By far the best sound system to use is ALSA. 
- Can leave `pulseaudio` option on Linux in case anyone was using it.
- Upstream also recommends `avahi` which provides compatibility with Bonjour